### ISSUE TITLE
NullReferenceException when using "protected" keywords as variable names

### ISSUE THREAD

**Issue by Lizzy-Gallagher** [trullock/NUglify#80]
I am minifying some npm packages and came across a failure when using multiple "protected" keywords as variable names.

JavaScript:
 ```
var o = {
	"a": function(async) {
		async = 5;
	},

	"b": function(module) {
		module.exports = function () { };
	}
};
```

C#: 
`result = Uglify.Js(content, fileName: filePath);`

Stacktrace:
NullReferenceException: Object reference not set to an instance of an object.
at NUglify.JavaScript.Syntax.MemberExpression.GetFunctionGuess(AstNode target)   at NUglify.JavaScript.Syntax.BinaryExpression.GetFunctionGuess(AstNode target)   at NUglify.JavaScript.Visitors.AnalyzeNodeVisitor.GuessAtName(AstNode node)   at NUglify.JavaScript.Visitors.AnalyzeNodeVisitor.Visit(FunctionObject node)   at NUglify.JavaScript.Syntax.FunctionObject.Accept(IVisitor visitor)   at NUglify.JavaScript.Visitors.AnalyzeNodeVisitor.Visit(BinaryExpression node)   at NUglify.JavaScript.Syntax.BinaryExpression.Accept(IVisitor visitor)   at NUglify.JavaScript.Visitors.AnalyzeNodeVisitor.Visit(BlockStatement node)   at NUglify.JavaScript.Syntax.BlockStatement.Accept(IVisitor visitor)   at NUglify.JavaScript.Visitors.AnalyzeNodeVisitor.Visit(FunctionObject node)   at NUglify.JavaScript.Syntax.FunctionObject.Accept(IVisitor visitor)   at NUglify.JavaScript.Visitors.AnalyzeNodeVisitor.Visit(BlockStatement node)   at NUglify.JavaScript.Syntax.BlockStatement.Accept(IVisitor visitor)   at NUglify.JavaScript.Visitors.AnalyzeNodeVisitor.Visit(FunctionObject node)   at NUglify.JavaScript.Syntax.FunctionObject.Accept(IVisitor visitor)   at NUglify.JavaScript.Visitors.TreeVisitor.Visit(ObjectLiteralProperty node)   at NUglify.JavaScript.Visitors.AnalyzeNodeVisitor.Visit(ObjectLiteralProperty node)   at NUglify.JavaScript.Syntax.ObjectLiteralProperty.Accept(IVisitor visitor)   at NUglify.JavaScript.Visitors.TreeVisitor.Visit(AstNodeList node)   at NUglify.JavaScript.Syntax.AstNodeList.Accept(IVisitor visitor)   at NUglify.JavaScript.Visitors.TreeVisitor.Visit(ObjectLiteral node)   at NUglify.JavaScript.Visitors.AnalyzeNodeVisitor.Visit(ObjectLiteral node)   at NUglify.JavaScript.Syntax.ObjectLiteral.Accept(IVisitor visitor)   at NUglify.JavaScript.Visitors.TreeVisitor.Visit(VariableDeclaration node)   at NUglify.JavaScript.Visitors.AnalyzeNodeVisitor.Visit(VariableDeclaration node)   at NUglify.JavaScript.Syntax.VariableDeclaration.Accept(IVisitor visitor)   at NUglify.JavaScript.Visitors.TreeVisitor.Visit(VarDeclaration node)   at NUglify.JavaScript.Visitors.AnalyzeNodeVisitor.Visit(VarDeclaration node)   at NUglify.JavaScript.Syntax.VarDeclaration.Accept(IVisitor visitor)   at NUglify.JavaScript.Visitors.AnalyzeNodeVisitor.Visit(BlockStatement node)   at NUglify.JavaScript.Syntax.BlockStatement.Accept(IVisitor visitor)   at NUglify.JavaScript.JSParser.InternalParse()   at NUglify.JavaScript.JSParser.Parse(DocumentContext sourceContext)   at NUglify.Uglify.Js(String source, String fileName, CodeSettings codeSettings)   at UserQuery.Minifiy(String filePath)   at UserQuery.Main()   at System.Threading.ThreadHelper.ThreadStart_Context(Object state)   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)   at System.Threading.ThreadHelper.ThreadStart()

**Comment by trullock** [comment]
Technically `async` isnt a reserved word, because all possible usages as a language feature are uniquely parseable. However, you must be a sadist to use such terms as variable names.

This might be hard to fix, depending on how well the async parsing has been implemented.

We'll try though. PRs welcome!

**Comment by trullock** [comment]
ok I've pushed a fix for the case you've found here, but there are other bugs remaining, namely

```
function (async) {
        async();
    }
```

i.e. if you overload `async` as an identifier, and then call it as a function (with or without arguments) the parser will not understand this line and will strip it, removing the invocation

I don't know how to fix this without some serious rewriting of the parser, which I can't be arsed to do for such a sadistic case.

**Comment by trullock** [comment]
Going to close this issue as its fixed, and i'll open a new one specific to the remaining case.

**Comment by trullock** [comment]
Published 1.6.6 with a fix for this in
